agent/fw_slim.c

```diff
@@ -97,6 +98,60 @@ NR_PHP_WRAPPER(nr_slim3_4_route_run) {
 }
 NR_PHP_WRAPPER_END
 
+static char* nr_slim_version(zval* app) {
+  char* retval = NULL;
+  zval* version = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(app)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application object is invalid",
+                     __func__);
+    return NULL;
+  }
+
+  ce = Z_OBJCE_P(app);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application has NULL class entry",
+                     __func__);
+    return NULL;
+  }
+
+  version = nr_php_get_class_constant(ce, "VERSION");
+  if (NULL == version) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application does not have VERSION",
+                     __func__);
+    return NULL;
+  }
+
+  if (nr_php_is_zval_valid_string(version)) {
+    retval = nr_strndup(Z_STRVAL_P(version), Z_STRLEN_P(version));
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: expected VERSION be a valid string, got type %d",
+                     __func__, Z_TYPE_P(version));
+  }
+
+  nr_php_zval_free(&version);
+  return retval;
```

Comment:
This fragment of code is repeated 3 times: [here](https://github.com/newrelic/newrelic-php-agent/blob/13b4a7e7bf8ef2f3bdcaab3f138d42cfb6e586f3/agent/fw_laravel.c#L489-L522), [here](https://github.com/newrelic/newrelic-php-agent/blob/13b4a7e7bf8ef2f3bdcaab3f138d42cfb6e586f3/agent/lib_guzzle6.c#L346-L380) and here](https://github.com/newrelic/newrelic-php-agent/blob/13b4a7e7bf8ef2f3bdcaab3f138d42cfb6e586f3/agent/lib_predis.c#L636-L669). It should really be extracted into a helper function.

Reply:
Good idea, I have created a helper function in `php_agent.c` that retrieves the value for an object constant and have removed the duplicate code: [e2305af](https://github.com/newrelic/newrelic-php-agent/pull/775/commits/e2305af3450da3728052fa1ca02b2b402415ddcd)